### PR TITLE
Task sf/j92 add socketio for comments

### PR DESCRIPTION
**Pull request by Sayonarko**

### CODE REVIEW

**Review by JuliaDzyuba — DISMISSED** [review]

**Review by Denidzio — DISMISSED** [review]

File: backend/src/middlewares/socket-injector/socket-injector.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+const socketInjector = (io: Server) => (
+  req: Request,
+  _res: Response,
+  next: NextFunction,
+): void => {
+  req.io = io;
+  next();
+};
+
+export { socketInjector };
```
Comment (ghost) [inline backend/src/middlewares/socket-injector/socket-injector.middleware.ts]:
```suggestion
export { socketInjector };

```

File: backend/src/middlewares/middlewares.ts [hunk 2]
```diff
@@ -7,3 +7,4 @@ export { authentication } from './authentication/authentication.middleware';
 export { checkAuth } from './check-auth/check-auth.middleware';
 export { registration } from './registration/registration.middleware';
 export { checkUserOwner } from './check-user-owner/check-user-owner.middleware';
+export { socketInjector } from './socket-injector/socket-injector.middleware';
```
Comment (ghost) [inline backend/src/middlewares/middlewares.ts]:
```suggestion
export { injectSocket } from './inject-socket/inject-socket.middleware';
```

File: shared/src/common/enums/socket/socket-event.enum.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+enum SocketEvent {
+  CONNECTION = 'connection',
+  JOIN_ROOM = 'joinRoom',
+  LEAVE_ROOM = 'leaveRoom',
+  UPDATE_COMMENTS = 'updateComments',
+}
+
+export { SocketEvent };
```
Comment (ghost) [inline shared/src/common/enums/socket/socket-event.enum.ts]:
```suggestion
export { SocketEvent };

```

File: frontend/src/components/episode/episode.tsx [hunk 4]
```diff
@@ -32,8 +33,14 @@ const Episode: React.FC = () => {
   const hasUser = Boolean(user);
 
   useEffect(() => {
+    socketService.joinRoom(id);
+    socketService.getUpdatedComments(dispatch, episodeActions.updateComments);
```
Comment (ghost) [inline frontend/src/components/episode/episode.tsx]:
```suggestion
```
we don't want to do it directly in the components
consider using packages for the redux to do it in the middleware

**Review by ghost — CHANGES_REQUESTED** [review]

File: backend/src/middlewares/inject-socket/inject-socket.middleware.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Request, Response, NextFunction } from 'express';
+import { Server } from 'socket.io';
+
+const injectSocket  = (io: Server) => (
+  req: Request,
+  _res: Response,
+  next: NextFunction,
+): void => {
+  req.io = io;
+  next();
```
Comment (ghost) [inline backend/src/middlewares/inject-socket/inject-socket.middleware.ts]:
```suggestion
  return next();
```

File: frontend/src/middlewares/middlewares.ts [hunk 6]
```diff
@@ -1 +1,3 @@
 export { handleError } from './handle-error/handle-error.middleware';
+export { socketMiddleware } from './socket/socket.middleware';
```
Comment (ghost) [inline frontend/src/middlewares/middlewares.ts]:
```suggestion
export { socket } from './socket/socket.middleware';
```

File: frontend/src/middlewares/socket/socket.middleware.ts [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+import { ENV, SocketEvent, DataStatus } from 'common/enums/enums';
+import { Comment } from 'common/types/types';
+import { ActionType as EpisodeActionType } from 'store/episode/common';
+import { episode as episodeAction } from 'store/actions';
+
+type Next = (action: AnyAction) => void;
+
+const socket = io(ENV.SOCKET_SERVER);
+
+const socketMiddleware: Middleware = ({ dispatch }) => (next): Next => {
```
Comment (ghost) [inline frontend/src/middlewares/socket/socket.middleware.ts]:
```suggestion
const socket: Middleware = ({ dispatch }) => (next): Next => {
```

Comment (ghost) [inline frontend/src/middlewares/socket/socket.middleware.ts]:
```suggestion
const socket = io(ENV.SOCKET_SERVER);

const socketMiddleware: Middleware = ({ dispatch }) => (next) => {
```
we really need it? 🤔

File: frontend/src/store/store.ts [hunk 8]
```diff
@@ -11,7 +11,7 @@ import {
   recordAudio as recordAudioService,
   notification as notificationService,
 } from 'services/services';
-import { handleError as handleErrorMiddleware } from 'middlewares/middlewares';
+import { handleError as handleErrorMiddleware, socketMiddleware } from 'middlewares/middlewares';
```
Comment (ghost) [inline frontend/src/store/store.ts]:
```suggestion
import { handleError as handleErrorMiddleware, socket as socketMiddleware } from 'middlewares/middlewares';
```

**Review by ghost — CHANGES_REQUESTED** [review]

File: frontend/src/middlewares/socket/socket.middleware.ts (hunk 7 above)
Comment (Sayonarko) [reply frontend/src/middlewares/socket/socket.middleware.ts]: Yep. Without this type checks failed

**Review by ghost — APPROVED** [review]

**Pull request merged**